### DISCUSSION TITLE
Ability to delete all annotations simultaneously

### DISCUSSION THREAD

**Discussion by lyucos** [sumatrapdfreader/sumatrapdf#4090] [Ideas]
Hello! Thank you for this reader!
As a user I'd like to be able to select all or some annotations so that to delete them. This would be a very handy feature!

**Comment by GitHubRulesOK** [comment]
Normally when you delete attached annotations all it does is switch them from active and visible to inactive and thus invisible , so in reality not a true deletion they are still lurking inside the file but the act of adding the deletions also duplicates much of their content so even more file size increase thus more slowdown in loading. 
The best solution is revert to the reserve copy before annotations were added thus it will be way smaller and faster.

**Reply by lyucos** [reply]
Thank you for response. Could you show how to revert to reserve copy?

**Reply by GitHubRulesOK** [reply]
There was a discussion https://github.com/sumatrapdfreader/sumatrapdf/discussions/3935

You must use a hex editor and some good online but for privacy best use only an online running local in PC https://hexed.it/ or a PC standalone hex editor https://mh-nexus.de/downloads/HxDPortableSetup.zip

You need to verify the additions were simply that so you search for more than one EOF (end of file) if you are certain that was the "incremental method" you can work backwards to the time when the file had no annotations HOWEVER beware for other reasons the source file may have had more than one ending, so not as simple as chop ALL from the first end of file.

Once you find that target EOF KEEP the 1 or 2 bytes after it (0A or 0D 0A) which is expected line feed then delete all after that point. daunting at first so work on a COPY.

**Comment by GitHubRulesOK** [comment]
To delete all annotations from a PDF is basically the function of a PDF editor and several ways to do that as a command line on one or more files. I would suggest coherent cpdf is probably best but have written a modifiable test script for doing similar at https://github.com/GitHubRulesOK/SumatraPDF-Plus/blob/master/Plus/MuPDF/DelPdfAnnots.js
